### PR TITLE
modified the api so the creation of custom tennant will work

### PR DESCRIPTION
**Pull request by DavidRajnoha**
Fixed the tenants url to /master/api/providers
Added the entity_id param to Tenant resource, so the delete would work correctly

### CODE REVIEW

**Comment by DavidRajnoha** [comment]
No, there is not yet. I have begun working on it but do not have the environment set to run the tests. I Will talk with jsmadis to solve it.

File: tests/integration/conftest.py [hunk 1]
```diff
@@ -350,3 +352,27 @@ def backend_metric(backend, metric_params) -> Metric:
...
+    """
+    Fixture for getting the custom tenant.
+    """
+    resource = master_api.tenants.create(tenant_params)
+    yield resource
+    resource.delete()
+    # tenants are not deleted immediately, they are scheduled for the deletion
+    # the exists method returns ok response, even if the tenant is scheduled for deletion
+    # However, the deletion of the tenant fails, if already deleted
+    with pytest.raises(errors.ApiClientError):
```
Comment (DavidRajnoha) [inline tests/integration/conftest.py:367]: This check if the tenant has been deleted is not nice, but I have not found a better way.

File: tests/integration/test_integration_custom_tenant.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from .asserts import assert_resource
+
+
+def test_3scale_master_url_is_set(master_api, master_url, master_token):
+    assert master_url is not None
```
Comment (mkudlej) [inline tests/integration/test_integration_custom_tenant.py:5]: I believe check like this is not needed and `assert master_url` is enough. This is valid comment also for next asserts.

File: tests/integration/conftest.py (hunk 1 above)
Comment (mkudlej) [reply tests/integration/conftest.py:367]: @DavidRajnoha I think you should wait here for some time until `resouce.delete` returns exception.

Comment (mkudlej) [reply tests/integration/conftest.py:367]: Ok, so after chat with @DavidRajnoha we've agreed that this is best solution.

**Pull request merged**